cmd/serve/serve.go

```diff
@@ -0,0 +1,47 @@
+package serve
+
+import (
+	"context"
+
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+	"github.com/likecoin/likecoin-chain-tx-indexer/poller"
+	"github.com/likecoin/likecoin-chain-tx-indexer/rest"
+	"github.com/spf13/cobra"
+)
+
+var Command = &cobra.Command{
+	Use:   "serve",
+	Short: "Run the indexing service and expose HTTP API",
+	Run: func(cmd *cobra.Command, args []string) {
+		restConn, err := db.NewConnFromCmdArgs(cmd)
+		if err != nil {
+			logger.L.Panicw("Cannot connect to Postgres", "error", err)
+		}
+		defer restConn.Close(context.Background())
+		pollerConn, err := db.NewConnFromCmdArgs(cmd)
+		if err != nil {
+			logger.L.Panicw("Cannot connect to Postgres", "error", err)
+		}
+		defer pollerConn.Close(context.Background())
+		err = db.InitDB(pollerConn)
+		if err != nil {
+			logger.L.Panicw("Cannot initialize Postgres database", "error", err)
+		}
+		listenAddr, err := cmd.PersistentFlags().GetString("listen-addr")
+		if err != nil {
+			logger.L.Panicw("Cannot get listen address from command line parameters", "error", err)
+		}
+		lcdEndpoint, err := cmd.PersistentFlags().GetString("lcd-endpoint")
+		if err != nil {
+			logger.L.Panicw("Cannot get lcd endpoint address from command line parameters", "error", err)
+		}
+		go rest.Run(restConn, listenAddr, lcdEndpoint)
```

Comment:
Would be nice if we can check the node height vs db height here, and prompt user to import, or directly trigger an import here if the diff is too huge

Reply:
Prefer to reject serving if the height difference is too large, unless something like `--ignore-height-difference` is provided.
I think 3 days of height difference (expect ~5s per block) could be considered too large?